core/access_api.php

```diff
@@ -511,32 +511,24 @@ function access_has_bug_level( $p_access_level, $p_bug_id, $p_user_id = null ) {
 
 	$t_project_id = bug_get_field( $p_bug_id, 'project_id' );
 	$t_bug_is_user_reporter = bug_is_user_reporter( $p_bug_id, $p_user_id );
-	$t_access_level = access_get_project_level( $t_project_id, $p_user_id );
+	$t_bug_is_user_handler = bug_is_user_handler( $p_bug_id, $p_user_id );
 
-	# check limit_Reporter (Issue #4769)
-	# reporters can view just issues they reported
-	$t_limit_reporters = config_get( 'limit_reporters', null, $p_user_id, $t_project_id );
-	if( $t_limit_reporters && !$t_bug_is_user_reporter ) {
-		# Here we only need to check that the current user has an access level
-		# higher than the lowest needed to report issues (report_bug_threshold).
-		# To improve performance, esp. when processing for several projects, we
-		# build a static array holding that threshold for each project
-		static $s_thresholds = array();
-		if( !isset( $s_thresholds[$t_project_id] ) ) {
-			$t_report_bug_threshold = config_get( 'report_bug_threshold', null, $p_user_id, $t_project_id );
-			if( empty( $t_report_bug_threshold ) ) {
-				$s_thresholds[$t_project_id] = NOBODY;
-			} else {
-				$s_thresholds[$t_project_id] = access_threshold_min_level( $t_report_bug_threshold ) + 1;
-			}
-		}
-		if( !access_compare_level( $t_access_level, $s_thresholds[$t_project_id] ) ) {
+	# check special limits
```

Comment:
Proposal for an alternate implementation of this section with slightly better performance in some cases and slightly better readability
```php
	$t_project_id = bug_get_field( $p_bug_id, 'project_id' );
	
	# check special limits
	# limit reporters means users can view just issues they reported (Issue #4769))
	# limit handlers means users can only view issues assigned to them
	# Access is allowed if any of the two criteria is met.

	$t_bug_is_user_reporter = bug_is_user_reporter( $p_bug_id, $p_user_id );
	$t_limit_reporter = access_has_limited_view_for_reporter( $t_project_id, $p_user_id );
	if( $t_limit_reporter && !$t_bug_is_user_reporter ) {
		return false;
	}

	$t_bug_is_user_handler = bug_is_user_handler( $p_bug_id, $p_user_id );
	$t_limit_handler = access_has_limited_view_for_handler( $t_project_id, $p_user_id );
	if( $t_limit_handler && !$t_bug_is_user_handler ) {
		return false;
	}
```